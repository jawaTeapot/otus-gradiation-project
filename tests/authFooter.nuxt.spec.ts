import { describe, beforeEach, expect, it } from 'vitest'
import { setActivePinia, createPinia } from 'pinia'
import { mount } from '@vue/test-utils'
import component from '~/components/auth/authFooter.vue'

describe('authFooter component', () => {
  function buildWrapper (path = '/app') {
    const wrapper = mount(component, {
      global: {
        stubs: {
          NuxtLink: true // Стабируем NuxtLink
        },
        mocks: {
          $route: {
            path
          },
          $t: (msg: string) => msg
        }
      }
    })
    return wrapper
  }

  beforeEach(() => {
    setActivePinia(createPinia())
  })

  it('mounts without errors', () => {
    const wrapper = buildWrapper('/')
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
