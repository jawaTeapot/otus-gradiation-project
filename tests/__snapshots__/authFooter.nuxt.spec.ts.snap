// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`authFooter component > mounts without errors 1`] = `
"<footer>
  <div class="flex items-center justify-between p-2 md:p-3">
    <div class="flex items-center"><span class="mr-3">© 2013–2024</span><a class="el-link el-link--info is-underline" href="https://hwd.boomstream.com/ru/contact.html" target="_self">
        <!--v-if--><span class="el-link__inner">footer.logo</span>
        <!--v-if-->
      </a></div>
    <nuxt-link-stub to="[object Object]" type="primary"></nuxt-link-stub>
  </div>
</footer>"
`;
